Podcast/Player.swift

```diff
@@ -133,13 +137,19 @@ class Player: NSObject {
         // cleanup any previous AVPlayerItem
         pause()
         removeCurrentItemStatusObserver()
-        setCurrentEpisodeToPreviouslyPlayingEpisode()
 
+        if let listeningDuration = listeningDurations[episode.id] { // if we've played this episode before
+            currentTimeAt = listeningDuration.currentProgress
+        } else {
+            listeningDurations[episode.id] = ListeningDuration(id: episode.id, currentProgress: episode.currentProgress, percentageListened: 0, realDuration: nil)
+            currentTimeAt = episode.currentProgress
```

Comment:
Do we have two current progress's to update now?

Reply:
nope